docs/source/examples/private_utxo/private_utxo_transaction_family.rst

```diff
@@ -0,0 +1,411 @@
+    :language: protobuf
+    :lines: 22-
+
+UTXO
+----
+
+UTXO are stored as empty objects in state. The key uniquely identifies the
+UTXO. The value and details of the UTXO are described by the UtxoDocument held
+off chain.
+
+
+Addressing
+----------
+
+The PUTP uses the standard Sawtooth Merkle Addressing scheme of 70 byte hex
+encoded string, consisting of 6 characters of namespace and 64 characters of
+key.
+
+Asset Type Addresses
+++++++++++++++++++++
+
+AssetTypes are stored in the address type namespace: '4251e9', which is derived
+by hex encoding the first 3 bytes of the sha512 hash of the string
+'private_utxo.asset_types'.
+
+.. code-block:: python
+
+    >>> asset_namespace = hashlib.sha512("private_utxo.asset_types".encode())\
+    ... .hexdigest()[0:6]
+    >>> asset_namespace
+    '4251e9'
+
+AssetType addresses are derived from the contents of the create transaction.
+The name, issuer's hex encoded public key, and the nonce are concatinated and
+the sha512 hash is taken. The result is hex encoded and the first 64 bytes of
+the result are taken as the address.
+
+.. code-block:: python
+
+    >>> raw_key = name + issuer_public_key + nonce
+    >>> address = hashlib.sha512(raw_key.encode()).hexdigest()[0:64]
+
+
+Holding Addresses
++++++++++++++++++
+
+Holdings are stored in the address type namespace: 'f1fc42', which is derived by
+hex encoding the first 3 bytes of the sha512 hash of the string
+'private_utxo.holdings'.
+
+.. code-block:: python
+
+    >>> holdings_namespace = hashlib.sha512("private_utxo.holdings".encode())\
+    ... .hexdigest()[0:6]
+    >>> holdings_namespace
+    'f1fc42'
+
+Holding addresses are derived from the participants public key. The sha512 hash
+is taken of the participants hex encoded public key. The result is hex encoded
+and the first 64 bytes of the result are taken as the address.
+
+.. code-block:: python
+
+    >>> address = hashlib.sha512(public_key.encode()).hexdigest()[0:64]
+
+.. _utxo_addresses:
+
+UTXO Addresses
+++++++++++++++
+
+UTXOs are stored in the address type namespace: 'cbefaf', which is derived by
+hex encoding the first 3 bytes of the sha512 hash of the string
+'private_utxo.utxo'.
+
+.. code-block:: python
+
+    >>> utxo_namespace = hashlib.sha512("private_utxo.utxo".encode())\
+    ... .hexdigest()[0:6]
+    >>> utxo_namespace
+    'cbefaf'
+
+UTXOs addresses are derived from the UtxoDocument. The sha512 hash is taken of
+the UtxoDocument. The result is hex encoded and the first 64 bytes of the
+result are taken as the address.
+
+.. code-block:: python
+
+    >>> address = hashlib.sha512(utxo_document).hexdigest()[0:64]
+
+
+Transactions
+============
+
+Transaction Headers
+-------------------
+
+The settings for the common Sawtooth Transaction header fields.
+
+.. code-block:: javascript
+
+    {
+        family_name: "sawtooth_private_utxo"
+        family_version: "1.0"
+        encoding: "application/protobuf"
+    }
+
+Inputs and Outputs
+------------------
+
```

Comment:
Do we need to specify a deterministic ordering of inputs and outputs. For example multiple inputs or multiple outputs are listed in asscending order of hex address? Goal is to make sure that the hash of the concatenated I/Os is consistent.